### PR TITLE
test: add more tests about selecting resources

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

- When matching cluster-scoped resources are updated (appearance/updated/disappearance)
     - should place and remove resources from clusters accordingly
     - should update CRP status accordingly
- When namespace-scoped resources in a matching namespace are updated (appearance/updated/disappearance)
     - should place and remove resources from clusters accordingly
     - should update CRP status accordingly

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: test/e2e/setup_test.go [hunk 1]
```diff
@@ -88,6 +89,13 @@ var (
 	ignoreConditionLTTReasonAndMessageFields = cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "Reason", "Message")
 	ignoreAgentStatusHeartbeatField          = cmpopts.IgnoreFields(clusterv1beta1.AgentStatus{}, "LastReceivedHeartbeat")
 	ignoreNamespaceStatusField               = cmpopts.IgnoreFields(corev1.Namespace{}, "Status")
+
+	crpStatusCmpOptions = cmp.Options{
+		cmpopts.SortSlices(lessFuncCondition),
```
Comment (ryanzhang-oss) [inline test/e2e/setup_test.go:94]: where were these function used before? wouldn't golang complain about it if there is not used before?

File: test/e2e/placement_selecting_resources_test.go [hunk 2]
```diff
@@ -138,3 +142,453 @@ var _ = Describe("creating CRP and selecting resources by label", Ordered, func(
 		Eventually(finalizerRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove controller finalizers from CRP %s", crpName)
 	})
 })
+
+var _ = Describe("validating CRP when cluster-scoped resources become selected after the updates", Ordered, func() {
```
Comment (ryanzhang-oss) [inline test/e2e/placement_selecting_resources_test.go:146]: There are a lot of similarities between those cases. Would it make sense to share some code between them?

Comment (ryanzhang-oss) [reply test/e2e/placement_selecting_resources_test.go:146]: i.e. put the delete to afterAll and share one afterAll.

Comment (ryanzhang-oss) [reply test/e2e/placement_selecting_resources_test.go:146]: Also, did we place any cluster scoped resource (not namespace)?

Comment (ryanzhang-oss) [inline test/e2e/placement_selecting_resources_test.go:146]: will we see some failed cases (like failed to appy CR)?

File: test/e2e/setup_test.go (hunk 1 above)
Comment (zhiying-lin) [reply test/e2e/setup_test.go:94]: they're used in https://github.com/Azure/fleet/blob/main/test/e2e/actuals_test.go#L105.

The IDE will give the hints.

File: test/e2e/placement_selecting_resources_test.go (hunk 2 above)
Comment (zhiying-lin) [reply test/e2e/placement_selecting_resources_test.go:146]: will do in next PR.

Comment (zhiying-lin) [reply test/e2e/placement_selecting_resources_test.go:146]: I tried this structure,

var _ = Describe(..., func (){
         BeforeAll()
         AfterAll()
         var _ = Describe("creating CRP and selecting resources by label", Ordered, func() {
                 It()
                It()
         })
})

However, ginkgo does not allow to put another layer on the top of the "ordered" spec. :( Have already created the cleanupxx related func to share the code.

Comment (zhiying-lin) [reply test/e2e/placement_selecting_resources_test.go:146]: ` did we place any cluster scoped resource (not namespace)?`

not yet, could add another test case in the next PR

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**